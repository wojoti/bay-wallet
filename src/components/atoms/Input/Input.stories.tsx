import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {Input} from 'components/atoms';

export default {
  title: 'Atoms/Input',
  component: Input,
  argTypes: {onChange: {action: 'onChange'}},
} as ComponentMeta<typeof Input>;
const Template: ComponentStory<typeof Input> = args => <Input {...args} />;

export const Username = Template.bind({});
Username.args = {type: 'username', testId: ''};

export const Password = Template.bind({});
Password.args = {type: 'password', testId: ''};
