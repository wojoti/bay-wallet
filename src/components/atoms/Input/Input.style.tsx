import {StyleSheet} from 'react-native';
import {primary} from 'theme/colors';
import {defaultWeight, font} from 'theme/fonts';

const style = StyleSheet.create({
  wrapper: {
    width: '100%',
    height: 65,
    borderWidth: 1,
    borderRadius: 8,
    borderColor: primary,
    fontFamily: font,
    fontWeight: defaultWeight,
    paddingLeft: 30,
    fontSize: 16,
  },
});
export default style;
