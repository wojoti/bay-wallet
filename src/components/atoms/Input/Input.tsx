import {PropsWithChildren} from 'react';
import {StyleProp, TextInput, TextStyle} from 'react-native';
import style from './Input.style';

export type InputProps = PropsWithChildren<{
  accessibilityLabel: string;
  accessibilityHint: string;
  onChange: (value: string) => void;
  value: string;
  testId: string;
  safe?: boolean;
  placeholder?: string;
  customStyle?: StyleProp<TextStyle>;
}>;

const Input = ({
  accessibilityLabel,
  accessibilityHint,
  onChange,
  value,
  testId,
  safe = false,
  placeholder,
  customStyle,
}: InputProps) => {
  return (
    <TextInput
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      style={[style.wrapper, customStyle]}
      testID={testId}
      onChangeText={onChange}
      autoComplete={'off'}
      autoCorrect={false}
      placeholder={placeholder}
      value={value}
      secureTextEntry={safe}
    />
  );
};

export default Input;
