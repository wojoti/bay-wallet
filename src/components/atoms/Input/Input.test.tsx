import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import Input, {InputProps} from './Input';

let value = '';
const handleChange = jest.fn(inputValue => (value = inputValue));
const props: InputProps = {
  onChange: handleChange,
  testId: 'test-input-id',
  accessibilityLabel: 'Username input',
  accessibilityHint: 'Type username here',
  safe: false,
  value: value,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<Input {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render input', () => {
  render(<Input {...props} />);
  const inputElement = screen.getByTestId('test-input-id');
  expect(inputElement).toBeOnTheScreen();
});

test('should handle text change with onChange', () => {
  render(<Input {...props} />);
  const inputElement = screen.getByTestId('test-input-id');
  expect(inputElement).toBeOnTheScreen();
  fireEvent.changeText(inputElement, 'test1234');
  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(value).toBe('test1234');
});
