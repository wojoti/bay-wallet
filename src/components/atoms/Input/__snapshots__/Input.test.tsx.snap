// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<TextInput
  accessibilityHint="Allow to insert username"
  accessibilityLabel="Login input field"
  autoComplete="username"
  autoCorrect={false}
  onChangeText={[MockFunction]}
  style={
    {
      "borderColor": "#363062",
      "borderRadius": 8,
      "borderWidth": 1,
      "fontFamily": "Poppins-Light",
      "fontSize": 16,
      "fontWeight": "400",
      "height": 65,
      "paddingLeft": 30,
      "width": "100%",
    }
  }
  testID="test-input-id"
/>
`;
